### PR TITLE
fix various issues with limiting streams

### PR DESCRIPTION
**Pull request by SteffenDE**
https://github.com/phoenixframework/phoenix_live_view/commit/997f15cadc86703134b1755f1fa2a3808ff91b2a introduces an issue where limits were not correctly enforced when adding items in bulk, this PR addresses this.

It also fixes an issue with limits being applied differently on resets, because we previously did not remove stream elements when they were being re-added later. I think this was initially introduced to avoid issues with live components not being correctly rendered because of the PHX_SKIP optimization. To counteract this, we now remove all elements on reset to force them being added in the correct order, but also store the live component DOM nodes to correctly restore them.

References #2686.

### CODE REVIEW

File: assets/js/phoenix_live_view/dom_patch.js [hunk 1]
```diff
@@ -340,6 +322,11 @@ export default class DOMPatch {
 
   removeStreamChildElement(child){
     if(!this.maybePendingRemove(child)){
+      if(child.getAttribute(PHX_COMPONENT) && this.streamInserts[child.id]){
```
Comment (SteffenDE) [inline assets/js/phoenix_live_view/dom_patch.js:325]: maybe there's another way to do this?

File: lib/phoenix_live_view.ex [hunk 2]
```diff
@@ -1819,7 +1820,7 @@ defmodule Phoenix.LiveView do
         Enum.reduce(items, new_socket, fn item, acc -> stream_insert(acc, name, item, opts) end)
 
       %{} ->
-        config = get_in(streams, [:__configured__, name]) || []
+        config = get_in(streams, [:__configured__, name]) || opts
```
Comment (SteffenDE) [inline lib/phoenix_live_view.ex:1823]: We didn't pass the limit when initially creating a stream

File: lib/phoenix_live_view/test/dom.ex
Comment (SteffenDE) [inline lib/phoenix_live_view/test/dom.ex:1]: this was an adventure

File: assets/js/phoenix_live_view/dom_patch.js (hunk 1 above)
Comment (SteffenDE) [reply assets/js/phoenix_live_view/dom_patch.js:325]: (comment refers to the streamComponentRestore "hack")

File: lib/phoenix_live_view.ex (hunk 2 above)
Comment (SteffenDE) [reply lib/phoenix_live_view.ex:1823]: I think there might be another problem here when using `stream_configure`. In that case the opts are always ignored, right? Will need to add a test.

Probably something like `Keyword.merge(opts, get_in(streams, [:__configured__, name]) || [])`

Comment (SteffenDE) [reply lib/phoenix_live_view.ex]: https://github.com/phoenixframework/phoenix_live_view/pull/3035/commits/d4bc58ac3d4bde26e1f89631348c4f3b4bc44103

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

**Pull request merged**